### ISSUE TITLE
False positive `Unreachable code` after polars.read_excel

### ISSUE THREAD

**Issue by Nicolas-Bechstedt** [pylint-dev/pylint#11436]
### Bug description

```python
# pylint: disable=missing-module-docstring
import polars as pl

pl.read_excel("abc.xlsx")
print()
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:5:0: W0101: Unreachable code (unreachable)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 6.67/10, +0.00)
```

### Expected behavior

```python
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Pylint version

```shell
pylint 4.0.8
astroid 4.0.3
Python 3.14.7 | packaged by Anaconda, Inc. | (main, Aug 14 2026, 10:12:46) [MSC v.1942 64 bit (AMD64)]
```

### OS / Environment

Windows 11, cmd with Miniconda

### Additional dependencies

```python
astroid @ file:///C:/Users/task_177020207818600/croot/astroid_1770202314802/work
colorama @ file:///C:/Users/dev-admin/buildout/croot/colorama_1759952627215/work
dill @ file:///C:/Users/task_JYXM7jesTBOrO_0/croot/dill_1783436189788/work
isort @ file:///C:/Users/task_178876595889809/croot/isort_1788766638296/work
mccabe @ file:///opt/conda/conda-bld/mccabe_1644221741721/work
mypy_extensions @ file:///C:/Users/task_178040666674677/croot/mypy_extensions_1780406779433/work
platformdirs @ file:///C:/Users/task_SZemcR1eQ3SLN_0/croot/platformdirs_1785312168310/work
polars @ file:///home/conda/feedstock_root/build_artifacts/bld/rattler-build_polars_1788957763/work/polars-python
polars-runtime-32 @ file:///D:/bld/bld/rattler-build_polars-runtime-32_1788957639/work/polars-runtime-32
pylint @ file:///home/conda/feedstock_root/build_artifacts/bld/rattler-build_pylint_1788027350/work
tomli @ file:///C:/Users/task_178965189789984/croot/tomli_1789652036250/work
tomlkit @ file:///C:/Users/task_aoE62VfSRayCS_0/croot/tomlkit_1784720806197/work
```

**Comment by Nicolas-Bechstedt** [comment]
Since I cannot reopen the previous issue regarding the same problem, I created a new one here.
Previous issue: https://github.com/pylint-dev/pylint/issues/10016

This was supposed to be fixed in https://github.com/pylint-dev/pylint/commit/4f6c2412097120587d6fc01fa0d344c49780f225, but the problem persists.

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening a new issue, I fail to reproduce on main but I can reproduce on 4.0.8 / 4.0.9. Git bisected to 4f6c24120, which is unreleased yet. I'm going to try releasing soon but we need some final python 3.15 features.
